### PR TITLE
Improved naming of applications and get vs export

### PR DESCRIPTION
**Pull request by GernotMaier**
PR to improve naming of applications and some functions.

Regarding issue #820: using find and grep, I was trying to find instances:

- changed one 'add' to 'export in model/model_parameter.py

On `make` vs `generate`:

- moved application make_regular_arrays.py go generate_regular_arrays

And while changing application names, several applications have been renamed to introduce a consistent and recognizable naming:

- moved application produce_array_config.py to generate_array_config.py
- moved applications for db to start with `db_`
- moved applications for simulation to `simulate_`

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -64,32 +64,32 @@ optional-dependencies."tests" = [
...
-scripts.simtools-compare-cumulative-psf = "simtools.applications.compare_cumulative_psf:main"
 scripts.simtools-convert-all-model-parameters-from-simtel = "simtools.applications.convert_all_model_parameters_from_simtel:main"
 scripts.simtools-convert-model-parameter-from-simtel = "simtools.applications.convert_model_parameter_from_simtel:main"
+scripts.simtools-db-add-file-to-db = "simtools.applications.db_add_file_to_db:main"
 scripts.simtools-db-add-model-parameters-from-repository-to-db = "simtools.applications.db_add_model_parameters_from_repository_to_db:main"
+scripts.simtools-db-add-value-from-json-to-db = "simtools.applications.db_add_value_from_json_to_db:main"
 scripts.simtools-db-get-array-layouts-from-db = "simtools.applications.db_get_array_layouts_from_db:main"
+scripts.simtools-db-get-file-from-db = "simtools.applications.db_get_file_from_db:main"
+scripts.simtools-db-get-parameter-from-db = "simtools.applications.db_get_parameter_from_db:main"
 scripts.simtools-derive-mirror-rnda = "simtools.applications.derive_mirror_rnda:main"
```
Comment (tobiaskleiner) [inline pyproject.toml:76]: derive_mirror_rnda and tune_psf is quite similar. Maybe both derive? Or we group everything (derivations, tuning) under `calculate` (with the trigger rates)?

**Review by tobiaskleiner — APPROVED** [review]
Thanks @GernotMaier, this greatly improves the naming. I have added another suggestion, how we could increase the unificaiton of the naming.

File: pyproject.toml (hunk 1 above)
Comment (GernotMaier) [reply pyproject.toml:76]: Very good suggestion: renamed tune_psf to derive_psf_parameters

Note the difference between deriving model parameters (which is different to calculating e.g., trigger rates), so I think having `calculate...` and `derive...` is consistent.

Comment (tobiaskleiner) [reply pyproject.toml:76]: Ok this sounds good. You can merge.

**Pull request merged**